### ISSUE TITLE
org-roam-db-build-cache breaks when indexing org files containing tramp links to hosts that are temporarily unavailable

### ISSUE THREAD

**Issue by cpbotha** [org-roam/org-roam#967]
#### Steps to Reproduce

Create org file with a link like `[[file:/scp:nonexistent.com:/tmp]]` and then run `M-x org-roam-db-build-cache`, or just save that file with org-roam-mode active.

You will then see something like:

```
Tramp: Opening connection for nonexistent.com using scp...
Tramp: Sending command ‘exec ssh   -o ControlMaster=auto -o ControlPath='tramp.%C' -o ControlPersist=no -e none nonexistent.com’
Tramp: Waiting for prompts from remote shell...
Tramp failed to connect.  If this happens repeatedly, try
    ‘M-x tramp-cleanup-this-connection’
```

In my case this happens because I link to files on hosts which are not always reachable, because they are behind a VPN.

I can also trigger this issue by activating `M-x org-roam-doctor` on the test case org file.

Why does org-roam need to follow these links?

#### Backtrace

This does not trigger an error and backtrace, but the org-roam indexing is interrupted. I can see this because most of my org files which used to be org-roamable before tonight's org-roam update are not anymore.

#### Expected Results

org-roam indexing should not trigger access to all embedded tramp links.

#### Actual Results

See above. Indexing is interrupted. org-roam not usable anymore.

### Environment

- Emacs: GNU Emacs 26.3 (build 2, x86_64-pc-linux-gnu, GTK+ Version 3.22.30) of 2019-09-16
- Framework: N/A
- Org: Org mode version 9.3.7 (9.3.7-14-gb2b587-elpa @ /home/cpbotha/.emacs.d/elpa/org-20200720/)
- Org-roam: 1.2.0

**Comment by myshevchuk** [comment]
> I can see this because most of my org files which used to be org-roamable before tonight's org-roam update are not anymore.

Do you by chance know when approximately the regression was introduced?

**Comment by cpbotha** [comment]
Going by @myshevchuk 's PR #970, I have isolated the lines in question. Indeed, by evalling the following in a test org file:

``` emacs-lisp
(setq path "/scp:nonexistent.com:/tmp")
(setq file-path (file-truename (buffer-file-name)))
(list (file-truename (expand-file-name path (file-name-directory file-path))))
```

... I get the same tramp-trying-to-open-the-link behaviour.

By evalling the sexps in the last line from inner to outer, I have further confirmed @myshevchuk 's suspicion that `file-truename` is the culprit.

Indeed, I only have to eval `(file-truename "/scp:nonexistent.com:/tmp")` to get the offending behaviour.

After that I patched in #970 and now org-roam does not error out anymore when I save an org file. :)

**Comment by myshevchuk** [comment]
@myshevchuk it is not clear yet, whether the offending behaviour is a regression introduced recently or it was there from the very beginning. From what you've written, I figured it was a regression. Can you confirm this?

**Comment by cpbotha** [comment]
I've been using org-roam for a while now. This new behaviour started after a MELPA package upgrade 2 days ago. I saw the org-roam database being upgraded from version 6 to 7, so the new behaviour must have started during the development period from v6 to v7 database.

It could of course also be due to interaction between the built-in function `file-truename` and my org package, which was also upgraded.

**Comment by jethrokuan** [comment]
Closed with #970 , tracking the longer-term issue at #972
